src/Blog/Archive/ArchiveController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Blog\Archive;
+
+use App\Controller;
+use App\Blog\Entity\Tag;
+use App\Blog\Tag\TagRepository;
+use Cycle\ORM\ORMInterface;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Yiisoft\Data\Paginator\OffsetPaginator;
+
+final class ArchiveController extends Controller
+{
+    private const POSTS_PER_PAGE = 3;
+    private const POPULAR_TAGS_COUNT = 10;
+
+    protected function getId(): string
+    {
+        return 'blog/archive';
+    }
+
+    public function index(ArchiveRepository $archiveRepo): Response
+    {
+        $output = $this->render(__FUNCTION__, [
```

Comment:
I'd use `index` instead of `__FUNCTION__`. It's cool but doesn't add up to clarity.